### PR TITLE
refactor(components/datetime): timepicker strict mode support

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/783/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/783/storybooks/ag-grid-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/783/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/783/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/783/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/783/playground/)

File: libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts [hunk 1]
```diff
@@ -51,13 +51,13 @@ export class SkyTimepickerComponent implements OnInit, OnDestroy {
   @Output()
   public selectedTimeChanged: EventEmitter<SkyTimepickerTimeOutput> = new EventEmitter<SkyTimepickerTimeOutput>();
 
-  public set disabled(value: boolean) {
-    this._disabled = value;
-    this.changeDetector.markForCheck();
+  public set disabled(value: boolean | undefined) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts]: This is not an input, so can remain as-is.

File: libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts [hunk 2]
```diff
@@ -120,18 +118,18 @@ export class SkyTimepickerComponent implements OnInit, OnDestroy {
     return moment(this.activeTime).minute() + 0;
   }
 
-  public set selectedTime(newTime: SkyTimepickerTimeOutput) {
+  public set selectedTime(newTime: SkyTimepickerTimeOutput | undefined) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts:121]: This is not an input, so can remain as-is.

File: libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts [hunk 3]
```diff
@@ -175,94 +173,106 @@ export class SkyTimepickerComponent implements OnInit, OnDestroy {
...
       .subscribe(() => {
-        this.changeDetector.markForCheck();
+        this.#changeDetector.markForCheck();
       });
   }
 
   public ngOnInit(): void {
     this.setFormat(this.timeFormat);
-    this.addKeydownListner();
+    this.#addKeydownListner();
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts]: While we're changing the name of this property, we can also fix its spelling. `addKeydownListner` 😎

File: libs/components/datetime/src/lib/modules/timepicker/timepicker.directive.ts [hunk 4]
```diff
@@ -242,30 +257,31 @@ export class SkyTimepickerInputDirective
...
       if (this.timeFormat === 'hh') {
         currentFormat = 'h:mm A';
       }
       if (this.timeFormat === 'HH') {
         currentFormat = 'H:mm';
       }
       if (typeof this.returnFormat === 'undefined') {
-        this.returnFormat = currentFormat;
+        // TODO: Remove this non-null when the typing of the `timeFormat` input is made stricter.
+        this.returnFormat = currentFormat!;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/timepicker/timepicker.directive.ts]: Can this be refactored to only respect the two time formats? Then you wouldn't need the non-null assertion.
```
const currentFormat = (this.timeFormat === 'HH') ? 'H:mm' : 'h:mm A';

if (typeof this.returnFormat === 'undefined') {
  this.returnFormat = currentFormat;
}
```

File: libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/datetime/src/lib/modules/timepicker/timepicker.component.ts:121]: This is set to the value of `#modelValue` within the directive which can be `undefined` so it needed updated.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**